### PR TITLE
Limit the number of partition files to be displayed for FileGroupsDisplay

### LINKED ISSUE #6358: Limit the number of partition files to be displayed for FileGroupsDisplay
### Is your feature request related to a problem or challenge?

When a sql needs to hit thousands of files, the content shown by the `FileGroupsDisplay` is too much. 

### Describe the solution you'd like

I propose to limit the number of partition files shown. 

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by yahoNanJing**
# Which issue does this PR close?

Closes #6358.

# Rationale for this change

When a sql needs to hit thousands of files, the content of thousands of partition files shown by the FileGroupsDisplay is too much.

# What changes are included in this PR?

Only at most 5 of them will be shown. Others will be replaced by `...`

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/core/src/physical_plan/file_format/mod.rs [hunk 1]
```diff
@@ -275,7 +286,9 @@ impl<'a> Display for FileGroupsDisplay<'a> {
         let mut first_group = true;
         let groups = if self.0.len() == 1 { "group" } else { "groups" };
         write!(f, "{{{} {}: [", self.0.len(), groups)?;
-        for group in self.0 {
+        // To avoid showing too many partitions
```
[the remark this answers is not in the document]
Comment (alamb) [reply datafusion/core/src/physical_plan/file_format/mod.rs:289]: 👍

**Review by alamb — APPROVED** [review]
This looks like a nice change to me -- thank you @yahoNanJing 

What do you think about showing all the groups when the user does  `explain verbose`?

It seem like seeing all the files may still be helpful

If you agree, we can add that feature as a follow on PR.

cc @crepererum  for your thoughts

**Comment by crepererum** [comment]
I agree that we shouldn't print out the entire payload as part of the normal explain, so :+1:. Adding this to a `EXPLAIN VERBOSE` output would be nice though.

File: datafusion/core/src/physical_plan/file_format/mod.rs [hunk 2]
```diff
@@ -297,6 +310,9 @@ impl<'a> Display for FileGroupsDisplay<'a> {
             }
             write!(f, "]")?;
         }
+        if self.0.len() > max_groups {
+            write!(f, ", ...")?;
+        }
         write!(f, "]}}")?;
         Ok(())
     }
```
Comment (alamb) [inline datafusion/core/src/physical_plan/file_format/mod.rs:318]: Is there any way we can write a test for this code? So we don't break it in the future?

**Review by alamb — COMMENTED** [review]
I think this is good to go other than it needs a test. Thank you @yahoNanJing

**Pull request merged**